WalletWasabi/Stores/BitcoinStore.cs

```diff
@@ -14,38 +10,32 @@
 namespace WalletWasabi.Stores
 {
 	/// <summary>
-	/// The purpose of this class is to safely and performantly manage all the Bitcoin related data
-	/// that's being serialized to disk, like transactions, wallet files, keys, blocks, index files, etc...
+	/// The purpose of this class is to safely and efficiently manage all the Bitcoin related data
+	/// that's being serialized to disk, like transactions, wallet files, keys, blocks, index files, etc.
 	/// </summary>
 	public class BitcoinStore
 	{
 		public BitcoinStore(
-			string workFolderPath,
-			Network network,
 			IndexStore indexStore,
 			AllTransactionStore transactionStore,
-			MempoolService mempoolService)
+			MempoolService mempoolService,
+			bool isMock = false)
 		{
-			WorkFolderPath = Guard.NotNullOrEmptyOrWhitespace(nameof(workFolderPath), workFolderPath, trim: true);
-			IoHelpers.EnsureDirectoryExists(WorkFolderPath);
-
-			Network = Guard.NotNull(nameof(network), network);
-			IndexStore = indexStore;
-			TransactionStore = transactionStore;
-			MempoolService = mempoolService;
-		}
-
-		/// <summary>
-		/// Special constructor used by the mock version.
-		/// </summary>
-		internal BitcoinStore()
-		{
-			TransactionStore = new AllTransactionStoreMock();
+			if (isMock)
```

Comment:
Can this be solved somehow in a different way? Interface, inherit and BitcoinStoreMock?

Reply:
The most straightforward way is to actually use IBitcoinStore interface instead of BitcoinStore.

Big disadvantage of `BitcoinStoreMock` class is that it's ONE type of mock. But you can mock `BitcoinStore` in a myriad of other ways. So I find it very inconvenient. 

To be clear, I don't like `isMock` too but I'm not sure I have too many options here. I would love to add `IBitcoinStore` interface but the PR would be even bigger and ... Well, I'm not sure. The other solution would be to simply remove `isMock` & Guards from the constructor with `#nullable enabled` :) What is your take on this?

Reply:
Nullable merged, so can you go in that direction? However, I am not sure that I understand your idea but I will see.

Reply:
With nullable merged, I would simply remove `isMock` & guards and rely on NRT, so that it's nicer than now.

Reply:
Sounds good.

Reply:
Removed.